src/libsystemd/sd-bus/sd-bus.c

```diff
@@ -3104,8 +3104,8 @@ static int bus_exit_now(sd_bus *bus) {
 
         log_debug("Bus connection disconnected, exiting.");
 
-        if (bus->event)
-                return sd_event_exit(bus->event, EXIT_FAILURE);
+        if (bus->event || event)
+                return sd_event_exit(bus->event ?: event, EXIT_FAILURE);
```

Comment:
maybe just use `event` here but up top do: 

```c
if (!event)
        event = bus->event;
```

feels better to me, because reduces the number of checks we need to do